src/services/async-component-resolver.ts

```diff
@@ -0,0 +1,23 @@
+import { Type, isFunction } from '@angular/core/src/facade/lang';
+import { Injectable, ComponentFactory, ComponentResolver } from '@angular/core';
+
+/**
+ * Component resolver that can load components lazily
+ *
+ * TODO: Update to use AppModule lazy loading
+ */
+@Injectable()
+export class AsyncComponentResolver implements ComponentResolver {
+  constructor(private _resolver: ComponentResolver) {}
+
+  resolveComponent(componentType: string|Type): Promise<ComponentFactory<any>> {
+    if (isFunction(componentType) && (<any>componentType).name === 'component') {
```

Comment:
The `ComponentResolver` uses the value of the `component` to determine how to resolve the component. Since the `component` is a function, I'm using the `name`, which for a non lazy-loaded component would be `NotFoundPage`, but for a lazy loaded component it will always be `component` in this case. I didn't test against minification since its not in the webpack config.